src/index.js

```diff
@@ -329,6 +347,63 @@ export default class AuthnWidget {
     }
   }
 
+  postEmptyAuthentication() {
+    this.store.dispatch('POST_FLOW', 'authenticate', '{}');
+  }
+
+  registerMfaEventHandler() {
+    Array.from(document.querySelectorAll('[data-mfa-selection]'))
+      .forEach(element => element.addEventListener('click', this.handleMfaDeviceSelection));
+  }
+
+  handleMfaDeviceSelection(evt) {
+    evt.preventDefault();
+    let source = evt.currentTarget;
+    if (source) {
+      let deviceId = source.dataset['mfaSelection'];
+      console.log(deviceId)
+      let data = {
+        "deviceRef": {
+          "id": deviceId
+        }
+      };
+      this.store.dispatch('POST_FLOW', "selectDevice", JSON.stringify(data));
+    } else {
+      console.log("ERROR - Unable to dispatch device selection as the target was null");
+    }
+  }
+
+  registerMfaChangeDeviceEventHandler() {
+    Array.from(document.querySelectorAll('#changeDevice'))
```

Comment:
If we are expecting multiple changeDevice elements, I think we should make it a data element rather than an ID. I don't think its technically valid HTML to have multiple elements with the same ID on the page.

If only expecting one I think we can safely remove the Array.from and just do a querySelector call.

Reply:
this actually is just for the back button (change device) functionality. The changing of the devices uses data elements.

Reply:
Do you think we should change it to just look for one element, rather than multiple elements?

Reply:
yep, done!